compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3476,6 +3482,12 @@ boolean isAssignableToFiniteType(BType type, BLangLiteral literalExpr) {
             if (((BLangLiteral) memberLiteral).value == null) {
                 return literalExpr.value == null;
             }
+
+            // If the literal which needs to be tested is from finite type and the type of the any member literal
+            // is not the same type, the literal cannot be assignable to finite type.
+            if (literalExpr.isFiniteContext && memberLiteral.getBType().tag != literalExpr.getBType().tag) {
```

Comment:
```ballerina
type FloatOne 1.0;
type DecimalOne 1.0d;

public function main() {
    FloatOne floatOne = 1.0;
    DecimalOne _ = floatOne;
}
```
Why do we disallow the above scenario? Don't we need to allow that? @MaryamZi 
We allow the following scenario.
```ballerina
type DecimalOne 1.0d;

public function main() {
    DecimalOne _ = 1.0;

}
```

Reply:
For the singleton types, if the literal value do not have any syntactical identifiers, the value will belong to the first basic type in the order int, float, decimal. So the literal `1.0` belongs to the basic type `float` and the singleton value `floatOne` belongs to the type `FloatOne` cannot be assigned to the type `DecimalOne` since the numeric literal of the singleton type `DecimalOne` belongs to the basic type `decimal`. So we should not allow the assignability of `floatOne` to the type `DecimalOne` in the above case.

Reply:
Shall we add a test case for this

Reply:
Already added test cases for this https://github.com/ballerina-platform/ballerina-lang/pull/35151/files#diff-59c530e0b5b6b1a7a6b270e98583394104adbda742a6f36588da13b7f5c4ac35R250